src/app/cluster/details/external-cluster/external-machine-deployment-details/component.ts

```diff
@@ -130,12 +131,8 @@ export class ExternalMachineDeploymentDetailsComponent implements OnInit, OnDest
     this._unsubscribe.complete();
   }
 
-  getStatusColor(): string {
-    return ExternalMachineDeployment.getStatusIcon(this.machineDeployment);
-  }
-
-  getStatusMessage(): string {
-    return ExternalMachineDeployment.getStatusMessage(this.machineDeployment);
+  getHealthStatus(): HealthStatus {
```

Comment:
Instead of this method, can we store health status in a field when `machineDeployment` is loaded?